common/tools/dev-tool/src/commands/run/typecheck.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as path from "node:path";
+import { leafCommand, makeCommandInfo } from "../../framework/command";
+import { createPrinter } from "../../util/printer";
+import { resolveProject } from "../../util/resolveProject";
+import { Project } from "ts-morph";
+import { DiagnosticCategory } from "typescript";
+
+const log = createPrinter("typecheck");
```

Comment:
does that mean we could build scripts like `typecheck:sample: dev-tool typecheck --samples-dev`?

do you know how to include this in ci if we want to enable this checking for mgmt?

Reply:
I think we could use the `build:samples`, like `"build:samples": "dev-tool run typecheck --paths samples-dev/**/*.ts"

Reply:
build:samples gets run in one of our CI configurations.